src/js/stores/DCOSStore.js

```diff
@@ -441,7 +457,11 @@ class DCOSStore extends EventEmitter {
   }
 
   get taskLookupTable() {
-    return this.buildFlatServiceTree(this.serviceTree);
+    if (!this._flatServiceTree) {
```

Comment:
we don't invalidate `this._flatServiceTree`. Does that mean that after we've invalidated `_flatServiceTree` we still going to be returning stalled version of `this._flatServiceTree`?

Reply:
@nLight We do whenever service tree is changed. See the get serviceTree if statement

Reply:
Is there a scenario where you could have called one of the above methods that nulls out the serviceTree, but havent called `getServiceTree`, which nulls out flatServiceTree, before calling `getTaskLookupTable`? Or is it guaranteed that getting the lookup table will only happen after calling getservicetree?

Reply:
@nLight @jaxesn I understood your problem and fixed it 😇

Reply:
I think to make this simpler to reason, we could invalidate both serviceTree and flatServiceTree together on a  method like `clearServiceTree`. Then they are both rebuilt lazily when needed (when the getter is called, we build one, or both.